src/model/utils/modifyselection.js

```diff
@@ -172,7 +172,8 @@ function getCorrectWordBreakPosition( walker, isForward ) {
 			// should expand to : 'foofoo [bar<$text bold="true">bar</$text>] bazbaz'.
 			const nextNode = isForward ? walker.position.nodeAfter : walker.position.nodeBefore;
 
-			if ( nextNode ) {
+			// `nextNode` can be an inline element which is an invalid node in this case.
```

Comment:
More like:

> Scan only text nodes. Ignore inline elements (like `<softBreak>`).

Reply:
The "invalid" part of your comment is kinda confusing – invalid means e.g. that a node cannot exist in that place. Not a big deal, but it can be written better.